### PR TITLE
[17] - Integration tests fixed on cryptocurrency wallet controller

### LINKED ISSUE #17: Integration test fix

### PR DESCRIPTION
**Pull request by IvanSolaDelgado**
**Issue**
- Resolves #17 

**Solution**
- I have used the laravel testing cache to test the controller as we did with OpenWallet, I have deleted a test since it had to do with the request params validation, and I will do that in the next PR fixing the validation :)

### CODE REVIEW

File: tests/app/Infrastructure/Controller/GetsWalletCryptocurrenciesControllerTest.php [hunk 1]
```diff
@@ -68,11 +40,8 @@ public function ifWalletIdExistsGetCoins()
         ];
         $wallet = new Wallet($walletId);
 
-        $this->walletDataSource
-            ->expects("findById")
-            ->with($walletId)
-            ->andReturn($wallet);
-
+        Cache::shouldReceive('has')->andReturn(true);
+        ;
```
Comment (AsierAlba10) [inline tests/app/Infrastructure/Controller/GetsWalletCryptocurrenciesControllerTest.php:44]: ?????
```suggestion
        
```

**Review by AsierAlba10 — COMMENTED** [review]
Tests namess!!!!!!

File: tests/app/Infrastructure/Controller/GetsWalletCryptocurrenciesControllerTest.php (hunk 1 above)
Comment (IvanSolaDelgado) [reply tests/app/Infrastructure/Controller/GetsWalletCryptocurrenciesControllerTest.php:44]: Wow, so strange it hasn't messed with that 🤣

File: tests/app/Infrastructure/Controller/GetsWalletCryptocurrenciesControllerTest.php [hunk 2]
```diff
@@ -11,42 +11,14 @@
...
-
-        $response = $this->get('api/wallet/-5');
-        $response->assertBadRequest();
-    }
-
-    /**
-     * @test
-     */
-    public function ifWalletIdNotFoundThrowsError()
+    public function returnsNotFoundMessageWhenWalletDoesNotExist()
```
Comment (AsierAlba10) [inline tests/app/Infrastructure/Controller/GetsWalletCryptocurrenciesControllerTest.php:17]: You don´t need to start all the tests with returns. Think about what are you going to test.
If you´re going to test that a wallet was not found if wallet does not exists.
```suggestion
    public function walletIdWasNotFoundIfWalletDoesNotExists()
```

**Review by AsierAlba10 — APPROVED** [review]

**Pull request merged**